src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -288,7 +296,7 @@ private long getMaxClientLagInMs() {
     Object val =
         this.parameterMap.getOrDefault(
             MAX_CLIENT_LAG,
-            isIcebergMode ? MAX_CLIENT_LAG_ICEBERG_MODE_DEFAULT : MAX_CLIENT_LAG_DEFAULT);
+            isIcebergMode() ? MAX_CLIENT_LAG_ICEBERG_MODE_DEFAULT : MAX_CLIENT_LAG_DEFAULT);
```

Comment:
this defaulting already happened in the setParameterMap method, does it need to happen again over here?

Reply:
This will not be set when both props and parameterOverrides are null.